src/movies.js

```diff
@@ -1,16 +1,113 @@
+
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(movies) {
+    let directors = movies.map (function(movie){
+        return movie.director;
+    })
+    return directors;
+} 
+
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(movies){
+
+    let occurance = movies.filter(function(movie) {
+
+        if (movie.director != "Steven Spielberg"){
+            return false;
+            
+        }
+        if (movie.genre.indexOf("Drama")== -1 ) {
+            return false;
+        }
+
+        return true;
+    });
+
+    return occurance.length;
+     
+}
+
+
+howManyMovies(movies);
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(movies){
+    if (movies.length == 0) {
+        return 0;
+    }
+
+   let total = movies.reduce(function(acc, curr){
+       if(!curr.rate) {
+           return acc;
+       }
+
+        return acc + curr.rate;
+    
+    }, 0);
+
+    let avg = total/movies.length;
+    return avg;
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
+function dramaMoviesRate(movies){
+    let occurances = movies.filter(function(movie) {
+
+        if (movie.genre.indexOf("Drama")== -1 ) {
```

Comment:
Using indexOf is perfectly fine, but there is another method called .includes that returns a boolean if an instance of the string appears in the array. So that one would be even better for cases like this.